### DISCUSSION TITLE
lldb-vscode

### DISCUSSION THREAD

**Discussion by adam-fowler** [swiftlang/vscode-swift#614] [Q&A]
For anyone using the 1.6 pre-release on Windows. Have you used the debugger that is packaged with the swift toolchain? How does it compare to CodeLLDB?

@compnerd @tristanlabelle

**Comment by tristanlabelle** [comment]
Our first experience is that we very easily ran into an assertion failure (crashing on Windows). We got that fixed here: https://github.com/apple/swift/pull/68634 , and we're in the process of checking its usability (more news later today, hopefully). I had noticed some cosmetic oddities like optionals showing as something like `none nil` and pointer addresses showing as hex without the `0` prefix (`xDEADBEEF`), but we should be able to iterate on those easily enough.

We're invested using `lldb-vscode.exe` from the toolchain because CodeLLDB seems to be a dead-end due to LLDB API/ABI changes in recent Swift toolchains.

**Reply by adam-fowler** [reply]
The one thing I noticed is it doesn't seem to expand enums with associated types where the type is something more complex than a number or string. Have you seen this?

**Reply by tristanlabelle** [reply]
It sounds weakly familiar but I'll be able to say more next week, I spent little time with it due to the assert/crash. We're looking to switch ~10 engineers to using the new DAP so I expect we'll find a lot of things to improve.

**Comment by adam-fowler** [comment]
I'm considering merging the vscode-lldb changes into the main branch. Managing two release branches is a pain. Do you have any concerns?

**Reply by tristanlabelle** [reply]
One concern: I'm testing right now and I can't debug tests with the codelldb adapter because `XCTest.dll` is not found. Debugging an executable does work. I'm investigating...

**Reply by tristanlabelle** [reply]
I found it, it's not directly related to the 1.6 changes, but for us it's hitting at the same time because we are trying to upgrade to a new Swift toolchain from main with a bumped version number: https://github.com/swift-server/vscode-swift/pull/618

**Reply by tristanlabelle** [reply]
No concerns with merging into main. The issue I found is unrelated, though we should get the fix in a release as soon as possible.
